tests/test_wallet_wallet_recovery.py

```diff
@@ -0,0 +1,626 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Tests for wallet.wallet_recoverey_manager module."""
+
+import json
+import tempfile
+import uuid
+from pathlib import Path
+
+import pytest
+import requests
+from aea.crypto.base import Crypto
+from aea_ledger_ethereum import EthereumCrypto
+from dotenv import load_dotenv
+from eth_account.signers.local import LocalAccount
+from web3 import Account
+
+from operate.cli import OperateApp
+from operate.constants import ZERO_ADDRESS
+from operate.operate_types import Chain, LedgerType
+from operate.utils.gnosis import add_owner, remove_owner, swap_owner
+from operate.wallet.master import MasterWalletManager
+from operate.wallet.wallet_recovery_manager import (
+    RECOVERY_BUNDLE_PREFIX,
+    RECOVERY_OLD_OBJECTS_DIR,
+    WalletRecoveryError,
+)
+
+from tests.conftest import OPERATE_TEST, RUNNING_IN_CI, random_string
+
+
+load_dotenv()
+
+# TODO operate.ledger must be loaded after load_dotenv() due to RPC env vars.
+from operate.ledger import get_default_rpc  # noqa: E402
+
+
+LEDGER_TO_CHAINS = {LedgerType.ETHEREUM: [Chain.GNOSIS, Chain.BASE]}
+
+
+def tenderly_add_balance(
```

Comment:
can we move it to `conftest.py`?

Reply:
Moved there, perhaps would be useful to create a utils.py file (conftest should only be used for fixtures)